src/deleteIndex.ts

```diff
@@ -3,16 +3,19 @@ import { getPineconeClient } from "./pinecone.js";
 import { getEnv, validateEnvironmentVariables } from "./utils/util.js";
 
 config();
-const indexName = getEnv("PINECONE_INDEX");
 
-const run = async () => {
+export const deleteIndex = async () => {
+  const indexName = getEnv("PINECONE_INDEX");
   validateEnvironmentVariables();
   // Initialize the Pinecone client
   const pineconeClient = await getPineconeClient();
 
-  await pineconeClient.deleteIndex({
-    indexName,
-  });
+  await pineconeClient
```

Comment:
I don't love the use of `await` and `then` mixed together. I think you can just log right after, and wrap the whole thing with a `try/catch` statement.

Reply:
Good point, thanks. I refactored code into try/catch